app/models/debt_balance.rb

```diff
@@ -100,7 +100,18 @@ def self.search(search)
   private
 
   def start_pay_date
+    if debt_id.nil?
+      errors.add(:debt, 'must not be emtpy')
+      return
+    end
+
+    if due_date.nil?
+      errors.add(:due_date, 'must not be empty')
```

Comment:
I tested without it and it breaks right into "start_pay_date". How do I make sure that the validation for debt and due_date happens before start_pay_date starts running ? Can we define the order of the validations?